fullstop-jobs/src/main/java/org/zalando/stups/fullstop/jobs/ami/FetchAmiJob.java

```diff
@@ -0,0 +1,203 @@
+        this.allAccountIds = allAccountIds;
+        this.jobsProperties = jobsProperties;
+        this.violationService = violationService;
+        this.taupageNamePrefix = taupageNamePrefix;
+        this.taupageOwners = Stream.of(taupageOwners.split(",")).filter(s -> !s.isEmpty()).collect(toList());
+    }
+
+    @PostConstruct
+    public void init() {
+        log.info("{} initalized", getClass().getSimpleName());
+    }
+
+    @Scheduled(fixedRate = 60_000 * 60 * 4, initialDelay = -1) // ((1 min * 60) * 4) = 4 hours rate, 0 min delay
+    public void run() {
+        log.info("Running job {}", getClass().getSimpleName());
+        for (String account : allAccountIds.get()) {
+            for (String region : jobsProperties.getWhitelistedRegions()) {
+
+                try {
+
+                    log.info("Scanning EC2 instances to fetch AMIs {}/{}", account, region);
+
+                    DescribeInstancesResult describeEC2Result = getDescribeEC2Result(
+                            account,
+                            region);
+
+                    for (final Reservation reservation : describeEC2Result.getReservations()) {
+
+                        for (final Instance instance : reservation.getInstances()) {
+                            final Map<String, Object> metaData = newHashMap();
+                            final List<String> errorMessages = newArrayList();
+
+                            if (violationService.violationExists(account, region, EVENT_ID, instance.getInstanceId(), OUTDATED_AMI)) {
+                                continue;
+                            }
+
+                            Optional<Image> image = getAmiFromEC2Api(account, region, instance.getImageId()).get().stream().findFirst();
+                            Optional<Boolean> isTaupageAmi = image
+                                    .filter(img -> img.getName().startsWith(taupageNamePrefix))
+                                    .map(Image::getOwnerId)
+                                    .map(taupageOwners::contains);
+
+                            // will not check for all non taupage ami
+                            // or images with taupage as name but created from another owner
+                            if (!isTaupageAmi.orElse(false)) {
+                                continue;
+                            }
+
+                            DateTime now = DateTime.now();
+                            if (isTaupageTooOld(image.map(Image::getName), now)) {
+                                metaData.put("ami_owner_id", image.map(Image::getOwnerId).orElse(""));
+                                metaData.put("ami_id", image.map(Image::getImageId).orElse(""));
+                                metaData.put("ami_name", image.map(Image::getName).orElse(""));
+                                DateTime maxValidityTimeForAmi = now.minus(Days.days(60));
+                                errorMessages.add("Outdated AMI! Please use the latest version available." +
```

Comment:
I would not add a generic `errorMessage` field to the meta data, because it is implicitly given by the violation type. Instead I would add a field `expiration_date` to show the user when the AMI actually had expired.